src/pages/FacilityOrganization/components/FacilityOrganizationSelector.tsx

```diff
@@ -94,102 +83,91 @@ export default function FacilityOrganizationSelector(
       hasChildren: org.has_children,
     }));
   };
-
   const handleEdit = (level: number) => {
     const newLevels = selectedLevels.slice(0, level);
     setSelectedLevels(newLevels);
+
     if (newLevels.length > 0) {
       const lastOrg = newLevels[newLevels.length - 1];
       setSelectedOrganization(lastOrg);
       onChange(lastOrg.id);
     } else {
       setSelectedOrganization(null);
+      onChange("");
     }
   };
 
+  const renderOrganizationLevel = (level: number) => {
+    const orgList =
+      level === 0
+        ? getAllOrganizations?.results
+        : currentLevelOrganizations?.results;
+    const getDropdownLabel = () => {
+      if (level < selectedLevels.length) {
+        return selectedLevels[level].name;
+      }
+      return level === 0 ? t("select_department") : t("select_sub_department");
+    };
+
+    return (
+      <div className="group flex items-center gap-1.5">
+        {level > 0 && (
+          <CareIcon
+            icon="l-arrow-right"
+            className="h-3.5 w-3.5 text-gray-400 flex-shrink-0"
+          />
+        )}
+        <div className="flex-1 flex items-center gap-2">
+          <div className="flex-1">
+            <Autocomplete
+              value={selectedLevels[level]?.id}
+              options={getOrganizationOptions(orgList)}
+              onChange={(value) => handleLevelChange(value, level)}
+              placeholder={getDropdownLabel()}
+            />
+          </div>
+          {level > 0 && level < selectedLevels.length && (
```

Comment:
This is no longer necessary since we are making others as editable autocompletes.